### DISCUSSION TITLE
Hugo

### DISCUSSION THREAD

**Discussion by brentstewart** [JupiterBroadcasting/jupiterbroadcasting.com#14] [General]
You mentioned in "Office Hours" that you were considering website options and that Hugo was on the list.  Let me recommend Hugo.

I had a brief period of involuntary unemployment at the start of the pandemic and started a blog to keep myself busy and keep my name in the conversation.  I settled on using Hugo after watching Mike Dane's excellent intro (https://www.youtube.com/watch?v=qtIqKaDlqXo) and playing around.

I write markdown in VSCode and QA locally using _hugo server -D_.  When I'm satisfied I upload to Github (https://github.com/brentstewart/nextpertise), where Github Actions renders it to HTML and uploads to Render (https://render.com/docs/static-sites).  Render will handle static sites for free, take care of my cert, and distribute it to their CDN.  Zero attack surface.  Had some friends in India test and they said it was just as fast there as here.  I wrote about this on my blog (https://www.nextpertise.net/posts/210102_hugoafter6/).

Just for giggles, I also tested this using public buckets at Azure and AWS S3.  That worked fine and I could have automated it as well, but it lacks some of Render's features. 

Markdown through a CI process to a static site is super simple and let's me concentrate on the fun stuff.  I have an add-in that uses Github issues to drive  my web comments.  I understand JB has a more diverse set of requirements, but host those portions separately (each piece does one thing well).

Looking forward to hearing what you do.  Let me know if I can help!

**Comment by murphym18** [comment]
Thank you for your post and blog post, I enjoyed reading it. I've been looking into Hugo for a personal site too. Your thoughts about it are helpful. I'd also add that it's fun and easy to build Hugo from source. I made an arm build last week and it works great. I like how flexible it is.

As for deployment, I was thinking of deploying to [Cloud Flare Pages](https://pages.cloudflare.com/). It has some nice features. You give pages access to your repo  and it takes care of deploying when you push to your `main` branch. This would save me from needing to use GitHub actions (though I'll probably need that eventually).  I also like the generous free tier. You get 500 builds per month, unlimited requests and unlimited data. [All pricing is at the bottom of this page](https://pages.cloudflare.com/). I do have concerns with pages though. Hugo is super flexible and it's not clear if pages will let me build Hugo from source. It's not that big a deal because I could always move on to GitHub actions, if needed.

But I didn't know about render and it looks compelling too. Overall I want to pick a host that contributes to the community. 

I enjoyed reading your posts. Thanks

**Comment by gerbrent** [comment]
Converting into a discussion, alongside the other wonderful website options here: https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/discussions/8
